main/src/vocab/base_vocab.rs

```diff
@@ -24,14 +29,32 @@ pub(crate) fn swap_key_values<T: Clone, U: Hash + Eq + Copy>(
         .collect()
 }
 
+#[allow(dead_code)]
+#[derive(Deserialize)]
+pub struct SpecialTokens {
+    additional_special_tokens: HashSet<String>,
```

Comment:
How would the `additional_special_tokens` be used? Should a `HashMap` be used for a mapping of `token_name` to `token_value`?